sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1186,6 +1186,143 @@ def max_deflection(self):
         else:
             return None
 
+
+    def bending_stress(self, y, m):
+        """
+        Returns bending stress of any point(or layer) in the beam cross-section at a distance `y`
+        from the neutral axis.
+
+        Parameters
+        ==========
+        y : Sympifiable
+            A symbol or a value representing the vertical distance of the required point(or layer)
+            on the cross-section from the neutral axis.
+        m : Sympifiable
+            A symbol or a value representing the bending moment at the required cross-section of
+            the beam.
+
+        Example
+        =======
+        There is a beam of length 6 m with a u.d.l of 40 KN/m over its length. The beam has a rectangular
+        cross-section(0.3 x 0.6 m) with moment of inertia. The bending stress at a point
+        0.2 m above the neutral axis and at a cross-section which is 4 meters from the start of the beam
+        has to be determined.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols, Symbol
+        >>> E, I = symbols('E, I')
+        >>> R1, R2 = symbols('R1, R2')
+        >>> b = Beam(6, E, I)
+        >>> b.apply_load(R1, 0, -1)
+        >>> b.apply_load(40, 0, 0, end=6)
+        >>> b.apply_load(R2, 6, -1)
+        >>> b.solve_for_reaction_loads(R1, R2)
+        >>> moment = b.bending_moment()
+        >>> x = Symbol('x')
+        >>> moment = abs(moment.subs(x, 4))
+        >>> b.bending_stress(0.2, moment)
+        32.0/I
+        """
+        b_stress = (m*y)/self.second_moment
+        return b_stress
+
+
+    def max_bending_stress(self, y_max, m=None):
```

Comment:
Maybe I'm being stupid but why do I need to provide y_max to get the max bending stress? Should this function not find y_max for me?

Reply:
I do agree with you. Ideally, it should be able to calculate `y_max`. But the problem lies in the way `beam` module works.
A user needs to give 3 things to define a beam. These are:
i) `L` length of the beam
ii) `E` Elastic Modulus of the material of the beam
iii) `I` second moment of area 

But there is no data about the shape of the cross section given by the user. So `y_max` could have been calculated internally if we had the data of the shape of the cross section, which is not the case as of now. Same is the case for `Q` i.e. the first moment of area.

But I guess this is something which can be done seperately as mentioned by moorepants in
 [this comment](https://github.com/sympy/sympy/pull/15775#discussion_r247360006). Cross sectional input could be taken and `I`, `Q`, `y_max` etc. could be calculated accordingly.